File: src/test/java/net/mreunionlabs/tapgradle/generate/generator/WebXMLGeneratorTest.java
package net.mreunionlabs.tapgradle.generate.generator;

import net.mreunionlabs.tapgradle.generate.GeneratePluginExtension;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

/**
 * Created by abangkis on 05-Sep-16.
 */
public class WebXMLGeneratorTest {
    private GeneratePluginExtension ext;

    @BeforeMethod
    public void setUp() throws Exception {
        ext = new GeneratePluginExtension();
        ext.setPackageString("net.mreunionlabs.gen");
        ext.setJavaDir(".");
    }
    @Test
    public void testCreateFile() throws Exception {
        WebXMLGenerator webXMLGenerator = new WebXMLGenerator();
        webXMLGenerator.createFile(ext);
    }

}

File: src/test/kotlin/net/mreunionlabs/tapgradle/generate/generator/WebXmlGeneratorTest.kt
package net.mreunionlabs.tapgradle.generate.generator

import io.kotlintest.TestCase
import io.kotlintest.specs.StringSpec
import net.mreunionlabs.tapgradle.generate.GeneratePluginExtension
import org.gradle.api.Project
import org.gradle.testfixtures.ProjectBuilder

class WebXmlGeneratorTest : StringSpec(){
    private lateinit var project: Project
    private lateinit var ext: GeneratePluginExtension

    override fun beforeTest(testCase: TestCase) {
        project = ProjectBuilder.builder().build()
        ext = GeneratePluginExtension()
        ext.packageString = "net.mreunionlabs.test"
        ext.webAppDir = "gen/webapp"
    }

    init {
        "create web xml file test should run" {
            WebXmlGenerator().createFile(ext)
        }
    }
}